engine-plugins/model-plugins/src/main/scala/org/trustedanalytics/atk/engine/model/plugins/timeseries/ARIMAXPredictPlugin.scala

```diff
@@ -0,0 +1,104 @@
+import org.trustedanalytics.atk.domain.frame._
+import org.trustedanalytics.atk.domain.schema.{ Column, DataTypes }
+import org.trustedanalytics.atk.engine.frame.SparkFrame
+import org.trustedanalytics.atk.engine.model.Model
+import org.trustedanalytics.atk.engine.plugin.{ ApiMaturityTag, Invocation, PluginDoc, SparkCommandPlugin }
+import org.trustedanalytics.atk.scoring.models.ARIMAXData
+
+// Implicits needed for JSON conversion
+import spray.json._
+import org.trustedanalytics.atk.domain.DomainJsonProtocol._
+import org.trustedanalytics.atk.engine.model.plugins.timeseries.ARIMAXJsonProtocol._
+
+@PluginDoc(oneLine = "New frame with column of predicted y values",
+  extended = """Predict the time series values for a test frame, based on the specified
+x values.  Creates a new frame revision with the existing columns and a new predicted_y
+column.""",
+  returns = """A new frame containing the original frame's columns and a column *predictied_y*""")
+class ARIMAXPredictPlugin extends SparkCommandPlugin[ARIMAXPredictArgs, FrameReference] {
+
+  /**
+   * The name of the command.
+   *
+   * The format of the name determines how the plugin gets "installed" in the client layer
+   * e.g Python client via code generation.
+   */
+  override def name: String = "model:arimax/predict"
+
+  override def apiMaturityTag = Some(ApiMaturityTag.Alpha)
+
+  /**
+   * Number of Spark jobs that get created by running this command
+   * (this configuration is used to prevent multiple progress bars in Python client)
+   */
+
+  override def numberOfJobs(arguments: ARIMAXPredictArgs)(implicit invocation: Invocation) = 1
+
+  /**
+   * Get the predictions for observations in a test frame
+   *
+   * @param invocation information about the user and the circumstances at the time of the call,
+   *                   as well as a function that can be called to produce a SparkContext that
+   *                   can be used during this invocation.
+   * @param arguments user supplied arguments to running this plugin
+   * @return a value of type declared as the Return type.
+   */
+  override def execute(arguments: ARIMAXPredictArgs)(implicit invocation: Invocation): FrameReference = {
+    val frame: SparkFrame = arguments.frame
+    val model: Model = arguments.model
+
+    //Extracting the ARIMAXModel from the stored JsObject
+    val arimaxData = model.data.convertTo[ARIMAXData]
+    val arimaxModel = arimaxData.arimaxModel
+
+    require(arimaxData.xColumns.length == arguments.xColumns.length, "Number of columns for train and predict should be the same")
+
+    // Get vector or y values and matrix of x values
+    val (yVector, xMatrix) = ARXFunctions.getYandXFromFrame(frame.rdd, arguments.timeseriesColumn, arguments.xColumns)
+    val ts = new DenseVector[Double](yVector.toArray)
+
+    // Find the maximum lag and fill that many spots with nulls, followed by the predicted y values
+    val maxLag = max(arimaxModel.p, arimaxModel.q)
```

Comment:
Looks like this is no longer needed.  You don't have null spots when we have a lag, right?